### PR TITLE
[SYCL][CUDA] Generalised Sampler Basic Test

### PR DESCRIPTION
**Pull request by StuartDAdams**
This PR introduces a generic test for `sycl::sampler`, and moves the existing testing to the OpenCL-specific `sampler_ocl.cpp`.

### CODE REVIEW

File: sycl/test/basic_tests/sampler/sampler.cpp [hunk 1]
```diff
@@ -50,37 +48,10 @@ int main() {
...
-    cl_sampler ClSampler =
-        clCreateSampler(Queue.get_context().get(), true, CL_ADDRESS_REPEAT,
-                        CL_FILTER_LINEAR, &Err);
-#endif
-    // If device doesn't support sampler - skip it
-    if (Err == CL_INVALID_OPERATION)
-      return 0;
+  B = sycl::sampler(sycl::coordinate_normalization_mode::normalized,
+                    sycl::addressing_mode::repeat,
+                    sycl::filtering_mode::linear);
```
Comment (bader) [inline sycl/test/basic_tests/sampler/sampler.cpp:53]: It look like this test in failing on non-CUDA back-ends.
@smaslov-intel, @romanovvlad, FYI.

> die: piextKernelSetArgSampler: not implemented

@StuartDAdams, I suggest marking this test as expected to fail on non-CUDA back-ends until we implement `piextKernelSetArgSampler` there.

Comment (bader) [reply sycl/test/basic_tests/sampler/sampler.cpp:53]: At least LevelZero plug-in should be updated.

Comment (StuartDAdams) [reply sycl/test/basic_tests/sampler/sampler.cpp:55]: Just saw this message, I've just pushed a new commit that updates the test with "XFAIL: level0".

**Comment by bader** [comment]
@StuartDAdams, @smaslov-intel added `piextKernelSetArgSampler` support to Level0, so I think you can revert latest commit. Sorry for inconvenience.

File: sycl/test/basic_tests/sampler/sampler.cpp [hunk 2]
```diff
@@ -1,10 +1,10 @@
-// REQUIRES: opencl
-
 // RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out -L %opencl_libs_dir -lOpenCL
 // RUN: env SYCL_DEVICE_TYPE=HOST %t.out
 // RUN: %CPU_RUN_PLACEHOLDER %t.out
 // RUN: %GPU_RUN_PLACEHOLDER %t.out
 // RUN: %ACC_RUN_PLACEHOLDER %t.out
+// XFAIL: level0
+// "die: piextKernelSetArgSampler: not implemented"
```
Comment (bader) [inline sycl/test/basic_tests/sampler/sampler.cpp:7]:
```suggestion
```

**Comment by bader** [comment]
> @StuartDAdams, @smaslov-intel added `piextKernelSetArgSampler` support to Level0, so I think you can revert latest commit. Sorry for inconvenience.

I managed to revert it myself. :-)

**Pull request merged**